internal/infrastructure/kubernetes/proxy/testdata/gateway-namespace-mode/deployment.yaml

```diff
@@ -8,7 +8,7 @@ metadata:
     app.kubernetes.io/name: envoy
     gateway.envoyproxy.io/owning-gateway-name: gateway-1
     gateway.envoyproxy.io/owning-gateway-namespace: namespace-1
-  name: envoy-namespace-1-gateway-1-e2117e41
+  name: gateway-1
```

Comment:
Should we keep the resource hash?

Reply:
IIUC, the hash is used to avoid `ns-1/name-1` with `ns/1-name-1`, prefer not to that.

Reply:
it will be conflict.

Reply:
if not adding hash, maybe just keep the envoy prefix

Reply:
@envoyproxy/gateway-maintainers wdyt?

Reply:
what won't help, e.g. gateway name with gw, and there's a deployment named `envoy-gw`.

Reply:
@envoyproxy/gateway-maintainers PTAL before merging this.

Reply:
+1 with PRs approach